### DISCUSSION TITLE
Read aircraft waypoints from Python

### DISCUSSION THREAD

**Discussion by Pelochus** [paparazzi/paparazzi#3289] [Q&A] [answered]
I might be blind or something, but I can't find a way to properly read the waypoints from an aircraft without reading its XML file. I don't want to do this since I might change the waypoints from the GCS.

Isn't there any telemetry message to read the relative (to the HOME waypoint) position of a waypoint?

**Comment by gautierhattenberger** [comment, marked as answer]
You have a message called `WP_MOVED_XXX` that is sent by the aircraft. Depending on the type of firmware and waypoint, you may receive `WP_MOVED` (fixedwing, UTM format), `WP_MOVED` (rotorcraft, local frame), `WP_MOVED_LLA` (rotorcraft, global frame).
You can also read the message `WAYPOINT_MOVED` sent by ground tool `server`, in lat/lon format for all types of aircraft (this is the message used by the GCS).

**Reply by Pelochus** [reply]
Yes I'm aware of that, but that message is only sent when the waypoint is moved right? I need to read the waypoints whether they have been moved or not

**Reply by gautierhattenberger** [reply]
They are also periodically sent by telemetry, see:
https://github.com/paparazzi/paparazzi/blob/master/sw/airborne/modules/nav/waypoints.c#L37
https://github.com/paparazzi/paparazzi/blob/master/conf/telemetry/default_rotorcraft.xml#L16 
If it is not fast enough, you can reduce the period in the telemetry file

**Reply by Pelochus** [reply]
Thank you very much Gautier, I assumed too soon that it would only be sent by event (moving the WP).
